internal/provisioner/helm/controllers/bundledeployment_controller.go

```diff
@@ -353,12 +353,16 @@ func getChart(chartfs fs.FS) (*chart.Chart, error) {
 			return nil
 		}
 		if baseDir == "" {
+			// capture the chart directory path
 			baseDir = path
 			return nil
 		}
 		if d.IsDir() {
 			return nil
 		}
+		if len(path) <= len(baseDir)+1 {
+			return fmt.Errorf("wrong chart directory: %s", baseDir)
+		}
```

Comment:
What's the base directory in this context? Can we use something like `strings.TrimPrefix(path, baseDir)` on line 375 to avoid the panic instead?

Reply:
The base directory is the `hello-world`.  It must be trimmed before it put into the storage.
```
hello-world/
hello-world/README.md
hello-world/Chart.yaml
hello-world/.helmignore
hello-world/templates
hello-world/templates/_helpers.tpl
hello-world/templates/service.yaml
hello-world/templates/serviceaccount.yaml
hello-world/templates/NOTES.txt
hello-world/templates/deployment.yaml
hello-world/values.yaml
```
I'll add check if the `path` has `baseDir` prefix.  It it does't, it must be error.  I'll change to use the TrimPrefix, too.  
Thanks!

Reply:
oh I realize that I probably gave you the wrong command. I'm reading the code right, we just want the `Chart.yaml` file name vs. the `hello-world/Chart.yaml` path. In that case, you can just use `filepath.Base(path)` to remove the directory structure.

Reply:
Thanks! I can use the `Base` function for the good case but I would like to make it fail when there are 2 charts in it.  I think the `TrimPrix` can work here.

Reply:
Are two charts an invalid helm chart hierarchy? We could delegate validating that edge case to the chart validation logic after we load the chart from storage.

Reply:
Oh, I misunderstood the `Base`.   It just needs to trim the top directory.  The file under a subdirectory should have the subdirectory name.

Reply:
The chart is validated before it is put into the storage. https://github.com/operator-framework/rukpak/blob/9b50108774e885a5a44ff8c65ca7e1200d82a15b/internal/provisioner/helm/controllers/bundle_controller.go#L136
https://github.com/operator-framework/rukpak/pull/473#discussion_r936120837